### ISSUE TITLE
Error parsing macro input when using `pub(in path)` syntax.

### ISSUE THREAD

**Issue by olson-sean-k** [mcarton/rust-derivative#21]
Using `#[derive(Derivative)]` produces a parsing error when members of a decorated `struct` use the `pub(in path)` syntax. For example:

```rust
pub mod foo {
    #[derivative(Debug, Hash)]
    #[derive(Derivative)]
    pub struct Bar {
        #[derivative(Debug = "ignore", Hash = "ignore")]
        pub baz: u32;
        pub(in foo) qux: u32; // Not visible outside of `foo`.
    }
}
```

This will cause an error like this: `help: message: failed to parse macro input: ...`. Using `pub` or `pub(super)` seems to work without any errors.

**Comment by dtolnay** [comment]
I believe this would be fixed by updating to a newer Syn version.

**Comment by mcarton** [comment]
It is indeed fixed in #26.
